icu4c/source/i18n/rbnf.cpp

```diff
@@ -1133,8 +1133,17 @@ RuleBasedNumberFormat::format(const DecimalQuantity &number,
 
             // TODO this section should probably be optimized. The DecimalFormat is shared in ICU4J.
             NumberFormat *decimalFormat = NumberFormat::createInstance(locale, UNUM_DECIMAL, status);
+            if (decimalFormat == nullptr) {
```

Comment:
It would be cleaner to use LocalPointer<NumberFormat> decimialFormat(...
It gets rid of the "delete decimalformat;" lines.

Reply:
Thanks for taking a look at the changes Andy! :)
I've pushed a new commit that uses `LocalPointer` instead, which cleans up the manual delete lines.